provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/BlackDuckProviderDataAccessor.java

```diff
@@ -195,26 +205,35 @@ public void updateProjectAndUserData(Long providerConfigId, Map<ProviderProject,
-        BlackDuckResponsesTransformer blackDuckResponsesTransformer = new BlackDuckResponsesTransformer(blackDuckServicesFactory.getBlackDuckHttpClient(), blackDuckJsonTransformer);
-
-        int offset = pageNumber * pageSize;
-        HttpUrl projectsRequestUrl = blackDuckServicesFactory.getBlackDuckHttpClient().getBaseUrl().appendRelativeUrl(ApiDiscovery.PROJECTS_LINK.getPath());
-        BlackDuckRequestBuilder blackDuckRequestBuilder = requestFactory.createCommonGetRequestBuilder()
-                                                              .url(projectsRequestUrl);
 
         Predicate<ProjectView> searchFilter = alwaysTruePredicate -> true;
         if (StringUtils.isNotBlank(searchTerm)) {
             searchFilter = projectView -> StringUtils.containsIgnoreCase(projectView.getName(), searchTerm);
         }
 
-        PagedRequest pagedRequest = new PagedRequest(blackDuckRequestBuilder, offset, pageSize);
-        BlackDuckPageResponse<ProjectView> projectViewBlackDuckPageResponse = blackDuckResponsesTransformer.getSomeMatchingResponses(pagedRequest, ProjectView.class, searchFilter, pagedRequest.getLimit());
+        BlackDuckPageResponse<ProjectView> pageOfViews = retrieveBlackDuckPageResponse(blackDuckServicesFactory, ApiDiscovery.PROJECTS_LINK_RESPONSE, pageNumber, pageSize, searchFilter);
+
+        List<ProviderProject> foundProjects = convertBlackDuckProjects(pageOfViews.getItems(), blackDuckService);
+        int totalPageCount = (pageOfViews.getTotalCount() + (pageSize - 1)) / pageSize;
+        return new AlertPagedModel<>(totalPageCount, pageNumber, pageSize, foundProjects);
+    }
+
+    private AlertPagedModel<ProviderUserModel> retrieveUsersForProvider(ConfigurationModel blackDuckConfigurationModel, int pageNumber, int pageSize, String searchTerm) throws IntegrationException {
+        BlackDuckServicesFactory blackDuckServicesFactory = createBlackDuckServicesFactory(blackDuckConfigurationModel);
 
-        List<ProviderProject> foundProjects = convertBlackDuckProjects(projectViewBlackDuckPageResponse.getItems(), blackDuckService);
-        int totalPageCount = (projectViewBlackDuckPageResponse.getTotalCount() + (pageSize - 1)) / pageSize;
+        Predicate<UserView> searchFilter = userView -> StringUtils.isNotBlank(userView.getEmail());
+        if (StringUtils.isNotBlank(searchTerm)) {
+            searchFilter = searchFilter.and(userView -> StringUtils.containsIgnoreCase(userView.getEmail(), searchTerm));
+        }
+
+        BlackDuckPageResponse<UserView> pageOfViews = retrieveBlackDuckPageResponse(blackDuckServicesFactory, ApiDiscovery.USERS_LINK_RESPONSE, pageNumber, pageSize, searchFilter);
+
+        List<ProviderUserModel> foundProjects = pageOfViews.getItems()
+                                                    .stream()
+                                                    .map(UserViewV4::getEmail)
```

Comment:
Can we not use UserView here because it doesn't declare the getEmail method?

Reply:
This was a typo that I have now fixed. It still worked though because they both deserialized to the same thing.